src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component, Element, Prop, Watch, h } from '@stencil/core';
+
+const VALID_STATES = ['', 'success', 'warning', 'danger'];
+const VALID_SIZES = ['', 'sm', 'md', 'lg', 'xl'];
+
+@Component({
+  tag: 'chi-text-input',
+  styleUrl: 'text-input.scss',
+  scoped: true
+})
+
+export class TextInput {
+  @Element() el: HTMLInputElement;
+  /**
+   * To define state color of Text input
+   */
+  @Prop() state: string;
+  /**
+   * To add a left positioned icon
+   */
+  @Prop() leftIcon: string;
+  /**
+   * To add a right positioned icon
+   */
+  @Prop() rightIcon: string;
+  /**
+   * To define color of left icon
+   */
+  @Prop() leftIconColor: string;
+  /**
+   * To define color of right icon
+   */
+  @Prop() rightIconColor: string;
+  /**
+   * To define size of Text input
+   */
+  @Prop() size: string;
+  /**
+   * To define palceholder of Text input
+   */
+  @Prop() placeholder: string;
+  /**
+   * To define value of Text input
+   */
+  @Prop() value: string;
+  /**
+   * To disable Text input
+   */
+  @Prop() disabled: boolean;
+  /**
+   * To define -active, -hover, -focus statuses
+   */
+  @Prop() status: string;
```

Comment:
Is status a needed Prop for some reason I don't understand? In my opinion it would be an internal state that changes when the user interacts with the component. You may expose methods to access and modify the state, but not a Prop. 
I would add a prop called disabled that should work as any other disabled prop. 
Just curious: What is the difference between active and focus?

Reply:
Status prop was necessary to emulate `-hover` and `-focus` states for tests.
I removed it and created a method `setState()` to be able to define internal `State status`.
Thanks